src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    [EventSource(Name = "System.Net.Http")]
+    internal sealed class HttpTelemetry : EventSource
+    {
+        public static readonly HttpTelemetry Log = new HttpTelemetry();
+
+        private IncrementingPollingCounter? _requestsPerSecondCounter;
+        private PollingCounter? _startedRequestsCounter;
+        private PollingCounter? _currentRequestsCounter;
+        private PollingCounter? _abortedRequestsCounter;
+
+        private long _startedRequests;
+        private long _currentRequests;
+        private long _abortedRequests;
+
+        public static new bool IsEnabled => Log.IsEnabled();
+
+        // NOTE
+        // - The 'Start' and 'Stop' suffixes on the following event names have special meaning in EventSource. They
+        //   enable creating 'activities'.
+        //   For more information, take a look at the following blog post:
+        //   https://blogs.msdn.microsoft.com/vancem/2015/09/14/exploring-eventsource-activity-correlation-and-causation-features/
+        // - A stop event's event id must be next one after its start event.
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void RequestStart(string host, int port, string scheme, string path, string version)
+        {
+            Interlocked.Increment(ref _startedRequests);
+            Interlocked.Increment(ref _currentRequests);
+            WriteEvent(1, host, port, scheme, path, version);
+        }
+
+        [Event(2, Level = EventLevel.Informational)]
+        public void RequestStop()
+        {
+            Interlocked.Decrement(ref _currentRequests);
+            WriteEvent(2);
+        }
+
+        [Event(3, Level = EventLevel.Error)]
+        public void RequestAbort()
```

Comment:
The most common pattern I see (though not by a large margin) seems to be NounVerb where the Verb is written past tense, with the aggravating exception of 'Start' and 'Stop' which are almost always written in that imperative form. IMO past tense also seems more intuitive given that we are almost always recording something that happened in the recent past.

Reply:
I will say that ASP.NET guidelines also adopted NounVerb but used 'present' tense and not 'past' tense:
https://gist.github.com/anurse/af1859663ac91c6cf69c820cebe92303

>Use present-tense verbs (Timeout not TimedOut)
Use NounVerb form (ConnectionStart not StartConnection)

But I tend to agree with you about "past tense also seems more intuitive given that we are almost always recording something that happened in the recent past."

So, I'll change these to use past tense (except for 'Start' and 'Stop').

Reply:
@davidsh do you know where you found the link to Andrew's gist? It would be nice if we can move the guidance to a location where the team can collectively edit it and redirect whatever link you followed.

Reply:
I found it here originally: https://github.com/dotnet/aspnetcore/issues/4769#issue-390516127

Reply:
Thanks! I opened PR https://github.com/dotnet/aspnetcore/pull/23036 to move the content of the gist. Once it is moved we'll be able to propose changes via PR and update the old links.